### DISCUSSION TITLE
A function defined with the functions UI fails with a permission denied error.

### DISCUSSION THREAD

**Discussion by jonascript** [supabase/supabase#8139] [General]
When I create same function using provided code in the DB console it executes successfully. 

Realize this is "alpha preview" so this issue could resolve itself. 

Steps to Repro: 
1. /[DB ID]/database/functions > "Create new function" 
2. Use UI to create function ( https://supabase.com/docs/guides/auth/managing-user-data#advanced-techniques 
4. Sign up new email 

Expected result: 
Inserts new profile

Actual result:
Permission denied error.

**Comment by GaryAustin1** [comment]
You likely did not declare the function "security definer" type in the advanced settings for the function.   You will need to delete and redo it though to change them.
This is needed because a function running as the user will not have enough privileges to access the Auth schema.  "Security Definer" means the function will run with the privileges of the "creator".

**Reply by jonascript** [reply]
Thanks @GaryAustin1. I actually didn't even notice you could scroll down to those settings 🤦.
